### PR TITLE
Tests: use semantic admin locators

### LINKED ISSUE #151: Replace CSS-class locators in admin e2e tests with semantic Playwright locators
## Context

PR 149 (`Modernize FOSSE admin screens`) introduced new Playwright assertions that bind to presentation-layer CSS classes from the new admin UI primitives. Flagged during code review on the `fosse-admin-visual-overhaul` branch.

Affected files and assertions:

- `tests/e2e/onboarding-wizard.spec.ts:67` — "Destination step shows two destination cards" asserts `.fosse-wizard__card.fosse-admin-card`, `.fosse-card-header`, `.fosse-card-body`, `.fosse-card-footer`.
- `tests/e2e/onboarding-wizard.spec.ts:108` — skip/continue layout test asserts `.fosse-wizard__skip`.
- `tests/e2e/onboarding-wizard.spec.ts:562-586` — completion alignment geometry test asserts `.fosse-wizard__complete-card`, `.fosse-complete-icon`.
- Similar class-string assertions across the bluesky-provider and status-page specs.

These break silently on CSS rename without DOM change. Playwright's guidance is to prefer `getByRole` / `getByLabel` / `getByText` and to fall back to stable data attributes only when the DOM has no semantic affordance.

## Direction

Migrate to **semantic locators**, not `data-testid` / `data-fosse-*` attributes on PHP templates. Avoiding test-only attributes in production markup is the preferred direction.

Approach per call site:
- Sections with a heading or landmark role -> `getByRole('region', { name: ... })` or `getByRole('heading', { name: ... })`.
- Buttons and links -> `getByRole('button', { name: ... })` / `getByRole('link', { name: ... })`.
- Form fields -> `getByLabel(...)`.
- Body text / status copy -> `getByText(...)`.

Edge cases worth designing for:
- The completion geometry test (`onboarding-wizard.spec.ts:562-586`) calls `getBoundingClientRect()` on `.fosse-complete-icon`, which has no role. Either (a) anchor the geometry check on a sibling that does have a semantic role and read the icon's rect via JS without a Locator, or (b) accept that this single decorative element keeps its CSS-class hook and document why.
- The skip/continue layout test reads geometry on `.fosse-wizard__skip`; same trade-off.

## Companion cleanup

While migrating, look at the PHPUnit assertions that grew in PR 149 for the same reason — `fosse-status-badge` modifiers, `fosse-choice-card`, `fosse-domain-handle-panel fosse-callout`, `fosse-wizard__form`, etc. The same class-string-as-test-contract pattern shows up there. Semantic alternatives for PHP unit tests are narrower (no Playwright locators), but at minimum the tests should target the *behavior* the class signals (state badge text, presence of an actionable form, etc.) rather than the class string itself when a behavioral check is available.

## Scope

Test code only. Do not modify PHP template attributes.

### PR DESCRIPTION
**Pull request by RCowles**
## Summary
- Replace admin e2e CSS-class locators with role, label, text, heading, and semantic DOM anchors.
- Add e2e helpers for semantic computed-style checks and provider status cards.
- Update related PHP render assertions to verify behavior, copy, links, forms, and semantic structure instead of presentation classes.

Fixes #151

## Testing
- pnpm run format:check
- pnpm run lint
- composer run-script lint-php
- composer run-script test-php -- --filter 'AP_ProviderTest|Bluesky_ProviderTest|Onboarding_WizardTest|Setup_PageTest'
- pnpm exec playwright test tests/e2e/onboarding-wizard.spec.ts tests/e2e/bluesky-provider.spec.ts tests/e2e/status-page.spec.ts
- git diff --check

### CODE REVIEW

File: tests/e2e/onboarding-wizard.spec.ts [hunk 1]
```diff
@@ -254,100 +273,57 @@ test( 'Destination selection navigates to identity step', async ( {
...
-	await expect(
-		page.locator( '.fosse-address-preview[data-fosse-mode="actor_blog"]' )
-	).toBeHidden();
-	await expect(
-		page.locator( '[data-fosse-when="includes-blog"]' )
-	).toBeVisible();
-	await expect(
-		page.locator( '#fosse-wizard-blog-identifier' )
-	).toBeVisible();
+	await expect( page.getByLabel( 'Site handle' ) ).toBeVisible();
```
[the remark this answers is not in the document]
Comment (RCowles) [reply tests/e2e/onboarding-wizard.spec.ts:313]: Fixed in 575997c. The blog-only branch now asserts both combined-preview labels are hidden while the Site fediverse preview and Site handle are visible. PR checks are green.

File: tests/e2e/onboarding-wizard.spec.ts [hunk 2]
```diff
@@ -254,100 +273,57 @@ test( 'Destination selection navigates to identity step', async ( {
...
 	await expect(
-		page.locator( '.fosse-address-preview[data-fosse-mode="actor_blog"]' )
-	).toBeVisible();
-	await expect(
-		page.locator( '.fosse-address-preview[data-fosse-mode="blog"]' )
-	).toBeHidden();
-	await expect(
-		page.locator( '[data-fosse-when="includes-blog"]' )
+		page.getByText( 'As your site:', { exact: true } )
 	).toBeVisible();
```
[the remark this answers is not in the document]
Comment (RCowles) [reply tests/e2e/onboarding-wizard.spec.ts:322]: Fixed in 575997c. The Both branch now asserts the Site handle field remains visible while the actor_blog preview is shown. PR checks are green.

[the remark this answers is not in the document]
Comment (RCowles) [reply tests/e2e/onboarding-wizard.spec.ts:322]: Fixed in 575997c. The Both branch now also asserts the actor-only and blog-only preview labels are hidden, so duplicate preview containers would fail the test. PR checks are green.

File: tests/php/Admin/Bluesky_ProviderTest.php [hunk 3]
```diff
@@ -2688,14 +2679,15 @@ public function test_render_connection_actions_surfaces_pending_revert_note(): v
...
+		$button_position = strpos( $decoded, 'Disconnect Bluesky' );
 
-		$this->assertStringContainsString( 'Disconnecting will also restore alice.bsky.social as this account\'s Bluesky handle', html_entity_decode( $output, ENT_QUOTES, 'UTF-8' ) );
+		$this->assertStringContainsString( 'Disconnecting will also restore alice.bsky.social as this account\'s Bluesky handle', $decoded );
 		$this->assertStringContainsString( 'alice.bsky.social', $output );
 		$this->assertIsInt( $note_position );
-		$this->assertIsInt( $footer_position );
-		$this->assertLessThan( $footer_position, $note_position );
+		$this->assertIsInt( $button_position );
+		$this->assertLessThan( $button_position, $note_position );
```
[the remark this answers is not in the document]
Comment (RCowles) [reply tests/php/Admin/Bluesky_ProviderTest.php]: Clarified in 2932d85. The previous assertion was technically checking note_position < button_position because PHPUnit uses assertLessThan(expected, actual), but I changed it to assertGreaterThan($note_position, $button_position) so the intended ordering reads directly. PR checks are green.

**Review by kraftbj — APPROVED** [review]
